onnxruntime/core/framework/session_state.cc

```diff
@@ -1120,9 +1182,15 @@ Status SessionState::FinalizeSessionStateImpl(const std::basic_string<PATH_CHAR_
                   });
   }
 
+  SubgraphsKernelCreateInfoMaps subgraphs_kernel_create_info_maps;
+  std::unordered_map<std::string, std::reference_wrapper<const ExecutionProviders>> subgraphs_execution_providers;
+  AccumulateAllNestedSubgraphsKernelCreateInfoMaps(*this, "", 0, subgraphs_kernel_create_info_maps, subgraphs_execution_providers);
```

Comment:
Do the EPs ever differ across subgraphs? AFAIK we just pass through the list from the main graph.

Reply:
It's a const ExecutionProviders& in the various session states as that was simpler than adding stuff to retrieve the EPs from the top level session state. 

AFAIK there's no way for a user to specify a different set of EPs for a subgraph, and the partitioning logic runs using the single set of EPs (it has no knowledge of references to that in the session state instances for subgraphs).

Reply:
Thanks for the explanation. Reverted all the subgraph EP accumulation logic.